tests/test_settings.py

```diff
@@ -64,6 +66,23 @@ def test_save(self, env_file):
             "XANADU_CLOUD_TLS": "False",
         }
 
+    def test_save_unprintable(self, env_file):
+        """Tests that settings won't be saved to a .env file
+        if REFRESH_TOKEN contains non_printables.
+        Note blank space at end of token below.
+        """
+        settings = xcc.Settings(
+            REFRESH_TOKEN="j.w.t ", HOST="example.com", PORT=80, TLS=False
+        )
```

Comment:
Note that a space is a printable character:
```pycon
>>> " ".isprintable()
True
```
But a newline is not:
```pycon
>>> "\n".isprintable()
False
```
```suggestion
        settings = xcc.Settings(
            REFRESH_TOKEN="j.w.t\n", HOST="example.com", PORT=80, TLS=False
        )
```